apps/web/src/design-system/table/Table.tsx

```diff
@@ -91,8 +112,48 @@ export function Table<T extends object>({
             pageCount: Math.ceil(total / pageSize),
           }
         : {}),
+      ...(withSelection && {
+        autoResetSelectedRows: false,
+        initialState: { selectedRowIds: initialSelectedRows ?? {} },
+      }),
     } as any,
-    usePagination
+    usePagination,
+    useRowSelect,
+    (hooks) => {
+      if (!withSelection) {
+        return;
+      }
+
+      const selectionRow = {
+        id: 'selection',
+        Header: () => null,
+        Cell: ({ row, isLoading }) => {
+          if (isLoading) {
+            return <RadioSkeleton />;
+          }
+
+          const { checked } = row.getToggleRowSelectedProps();
+
+          return (
+            <Radio
+              checked={checked}
+              onChange={(e) => {
+                e.stopPropagation();
+
+                toggleAllRowsSelected(false);
+                row.toggleRowSelected(true);
+                onRowSelect?.(row);
+              }}
+              size="sm"
+              styles={{ radio: { margin: '0 !important', cursor: 'pointer' } }}
+            />
+          );
+        },
+        width: 30,
+        maxWidth: 30,
+      };
+      hooks.visibleColumns.push((visibleColumns) => [selectionRow, ...visibleColumns]);
+    }
```

Comment:
this way we can append the column that represents the "radio" for a selection functionality